keras_imaging/preprocessing/image.py

```diff
@@ -13,80 +13,24 @@
 class ImageGenerator(object):
     def __init__(
         self,
-        correct_distortion=False,
-        correct_uneven_illumination=False,
-        correct_vignetting=False,
-        desaturate=None,
-        equalize=None,
-        flip_horizontally=False,
-        flip_vertically=False,
-        preprocessing_function=None,
-        reduce_noise=None,
-        remove_chromatic_aberration=False,
-        rescale_intensity=None,
-        rotate=False,
-        shift_horizontally=False,
-        shift_vertically=False,
-        smooth=False
+        transforms=[],
+        standardizations=[],
+        reader_name = "skimage"
     ):
         """
-        :param correct_distortion:
+        :param transforms:
 
-        :param correct_uneven_illumination:
+        :param standardizations:
 
-        :param correct_vignetting:
+        :param reader_name:
 
-        :param desaturate:
-
-        :param equalize:
-
-        :param flip_horizontally:
-
-        :param flip_vertically:
-
-        :param reduce_noise:
-
-        :param remove_chromatic_aberration:
-
-        :param rescale_intensity:
-
-        :param rotate:
-
-        :param shift_horizontally:
-
-        :param shift_vertically:
-
-        :param smooth:
         """
-        self.correct_distortion = correct_distortion
-
-        self.correct_uneven_illumination = correct_uneven_illumination
-
-        self.correct_vignetting = correct_vignetting
-
-        self.desaturate = desaturate
-
-        self.equalize = equalize
-
-        self.flip_horizontally = flip_horizontally
-
-        self.flip_vertically = flip_vertically
-
-        self.preprocessing_function = preprocessing_function
-
-        self.reduce_noise = reduce_noise
-
-        self.remove_chromatic_aberration = remove_chromatic_aberration
+        assert isinstance(transforms,list)
+        assert isinstance(standardizations, list)
```

Comment:
You can remove this assertion since `standardizations` can be any collection (e.g. set or tuple).